dig.go

```diff
@@ -170,6 +176,78 @@ func (c *Container) provide(ctor interface{}, ctype reflect.Type) error {
 	return nil
 }
 
+func (c *Container) findConnections(n *node) (map[key]struct{}, error) {
```

Comment:
had to keep reminding myself what this function did during the course of this review.  Maybe a rename to something like `findAndValidateResults`?